docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,669 @@
+* [How to cross-compile for .NET Core and .NET Framework](#how-to-cross-compile-for-net-core-and-net-framework)
+* [How to test libraries on .NET Core](#how-to-test-libraries-on-net-core)
+* [How to create a NuGet Package with your Library](#how-to-create-a-nuget-package-with-your-library)
+
+## Prerequisites
+
+You must have .NET Core installed on your machine.  You have two options:
+
+* [DNX](http://aspnet.readthedocs.org/en/latest/getting-started/index.html), which is part of the ASP.NET 5 RC1 and is included in Visual Studio Update 1.
+
+* [.NET CLI](http://dotnet.github.io/getting-started/), which will be used by .NET Core and ASP.NET RC2 and will replace DNX.
+
+Either toolchain can be used to build and publish libraries today.
+
+The sections of this document dealing with .NET Framework versions or Portable Class Libraries (PCLs) need the .NET Framework installed.  This is only supported on Windows.
+
+To do this, [install the .NET Framework](http://getdotnet.azurewebsites.net).
+
+Additionally, if you wish to support older targets, you will need to install targeting/developer packs for older framework versions from the [target platforms page](http://getdotnet.azurewebsites.net/target-dotnet-platforms.html).  Refer to this table:
+
+| .NET Framework Version | Thing to download |
+| ---------------------- | ----------------- |
+| 4.6 | .NET Framework 4.6 Targeting Pack |
+| 4.5.2 | .NET Framework 4.5.2 Developer Pack |
+| 4.5.1 | .NET Framework 4.5.1 Developer Pack |
+| 4.5 | Windows Software Development Kit for Windows 8 |
+| 4.0 | Windows SDK for Windows 7 and .NET Framework 4 |
+| 2.0 and 3.5 | .NET Framework 3.5 SP1 Runtime (or Windows 8+ version) |
+
+## How to target .NET Core
+
+For the purposes of building a library, targeting ".NET Core" means targeting the .NET Platform Standard.  To see what that means exactly, see [.NET Platform Standard](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/standard-platform.md).
+
+Note that as of this time, `netstandardX.X` is not supported.  You'll have to use the `dotnet` monikers.  Here's a mapping:
+
+| dotnet | netstandard |
+|--------| ----------- |
+| dotnet51 | netstandard1.0 |
+| dotnet52 | netstandard1.1 |
+| dotnet53 | netstandard1.2 |
+| dotnet54 | netstandard1.3 |
+| dotnet55 | netstandard1.4 |
+
+
+The following is what you need to know about .NET Platform Standard targeting:
+
+You need to pick a version of `dotnetXX` (where `XX` is a version number) to add to your `project.json` file.
+
+1. You acn target the latest version - `dotnet55` - of the .NET Platform Standard, which will give you access to the most APIs.  `dotnet55` is supported by .NET Framework 4.6 (and higher) and .NET Core 1. 
+2. You can target any earlier version of the .NET Platform Standard to target earlier .NET platforms and versions.  Earlier .NET Platform Standard versions expose fewer APIs. Example targets:
+
+    - .NET Framework 4.6, 4.5.2, 4.5.1, or 4.5
```

Comment:
I'm thinking some kind of re-wording is needed here.  What I'm thinking is something along the lines of:

> The version of the .NET Platform Standard you pick will be a tradeoff between access to the newest APIs and ability to target more .NET platforms and Framework versions.

> * You can use the latest version of the .NET Platform Standard - `dotnet55` - which is for when you want access to the most APIs and don't want to worry about targeting anything other than .NET Core or .NET Framework 4.6 and higher.
> * You can use a lower version of the .NET Platform Standard to target earlier .NET platforms and versions.  The cost here is not having access to some of the latest APIs.  Example targets:
>    - .NET Framework versions 4.5.2, 4.5.1, or 4.5
> ...